src/main/java/com/minecolonies/blockout/util/Parsing.java

```diff
@@ -0,0 +1,170 @@
+package com.minecolonies.blockout.util;
+
+import net.minecraft.util.math.MathHelper;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.minecolonies.blockout.util.Log.getLogger;
+
+/**
+ * A utility class with method that handle parsing tasks.
+ */
+public class Parsing
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this class "final" or add a public constructor. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2974)